features/0809-w3c-data-integrity-credential-attachment/README.md

```diff
@@ -0,0 +1,321 @@
+
+```json
+{
+  "@id": "284d3996-ba85-45d9-964b-9fd5805517b6",
+  "@type": "https://didcomm.org/issue-credential/2.0/request-credential",
+  "comment": "<some comment>",
+  "formats": [
+    {
+      "attach_id": "5b38af88-d36f-4f77-bb7a-2f04ab806eb8",
+      "format": "didcomm/w3c-di-vc-request@v0.1"
+    }
+  ],
+  "~attach": [
+    {
+      "@id": "123",
+      "mime-type": "application/json",
+      "data": {
+        "base64": "<base64-encoded-json-attachment-content>",
+        "jws": {
+          "protected": "eyJhbGciOiJFZERTQSIsImlhdCI6MTU4Mzg4... (bytes omitted)",
+          "signature": "3dZWsuru7QAVFUCtTd0s7uc1peYEijx4eyt5... (bytes omitted)"
+        }
+      }
+    }
+  ],
+  "credentials~attach": [
+    {
+      "@id": "5b38af88-d36f-4f77-bb7a-2f04ab806eb8",
+      "mime-type": "application/json",
+      "data": {
+        "base64": "ewogICAgICAgICAgIkBjb250ZXogWwogICAgICAg...(clipped)...RNVmR0SXFXZhWXgySkJBIgAgfQogICAgICAgIH0="
+      }
+    }
+  ]
+}
+```
+
+##### Binding in Credential
+
+The issued credential should be bound to the holder by including the did in the credential as `credentialSubject.id` or `holder`.
+
+## Drawbacks
+
+- While it has a similar setup and structure compared to OpenID for Verifiable Credential Issuance, this attachment format only focuses on issuance of W3C Verifiable Credentials. Therefore another (but probably quite similar) attachment format needs to be defined to support issuance of non-W3C VCs
+- There is currently no way for an issuer or holder to indicate which cryptographic suites they support for the signature of the credential. Currently it's at the discretion of the issuer to decide which cryptographic suite to use. In a future (minor) version we can add an optional way for a) the issuer to indicate which cryptographic suites they support, and b) the holder to indicate which cryptographic suites they support.
+- There is currently no attachment format defined for a credential proposal. This makes it impossible for a holder to initiate the issuance of a credential using this attachment format. In a future (minor) version the proposal attachment format can be added.
+
+## Rationale and alternatives
+
+[RFC 0593: JSON-LD Credential Attachment](https://github.com/hyperledger/aries-rfcs/blob/main/features/0593-json-ld-cred-attach/README.md), [W3C VC API](https://w3c-ccg.github.io/vc-api/) allows issuance of credentials using only linked data signatures, while [RFC 0592: Indy Attachment](https://github.com/hyperledger/aries-rfcs/blob/main/features/0592-indy-attachments/README.md) supports issuance of AnonCreds credentials. This attachment format aims to support issuance of both previous attachment formats (while for AnonCreds it now being in the W3C model), as well as supporting additional features such as issuance W3C JWT VCs, credentials with multiple proofs, and cryptographic binding of the credential to the holder.
```

Comment:
> allows issuance of credentials using only linked data signatures

The goal of the VC API is shifting. We wanted to start with a concrete end-to-end flow using a narrow profile, but given that the VCDM v2.0 [now supports arbitrary credential formats](https://w3c.github.io/vc-data-model/#enveloped-verifiable-credentials), and given that those arbitrary credential formats can (theoretically) just slot into the VC API, it is possible to use the VC API to issue/verify/derive alternate credential formats. 

We still need to do a lot of work to make that clear in the specification, and we're behind on the VC API because all of our effort is going into completing VCDM v2.0 and Data Integrity v1.0 now. Once that's done (in the next couple of months), we'll shift our focus into opening the aperature for the VC API. @PatStLouis is engaged over there and so I expect that we'll be able to align given his engagement in multiple groups.

Reply:
The credential attachment outlined here is specific to didcomm issuance/requests. While the json-ld attachment was initially designed around the vc-api, this was back in 2019 and does not represent the current state of the specification. However, the vc-api will still have shortcomming when dealing with holder binding negotiations (such as requesting a holder link secret bind with the anoncreds-2023 crypto suite).

From what I see the attachment proposed removed the `options` field in favor of the following:
```
{
  "data_model_versions_supported": ["1.1", "2.0"],
  "binding_required": true,
  "binding_method": {
    "anoncreds_link_secret": {
      "nonce": "1234",
      "cred_def_id": "did:key:z6MkwXG2WjeQnNxSoynSGYU8V9j3QzP3JSqhdmkHc6SaVWoT/credential-definition",
      "key_correctness_proof": "<key_correctness_proof>"
    },
    "didcomm_signed_attachment": {
      "algs_supported": ["EdDSA"],
      "did_methods_supported": ["key", "web"],
      "nonce": "1234"
    }
  }
}
```

There is a lot of format negotiation included in the didcomm issuance rfc which is not really in line with the prescriptive approach of the vc-api model. This being said, these fields could be included as options in a vc-api workflow without issue.

The credential [omissions ](https://github.com/hyperledger/aries-rfcs/tree/main/features/0809-w3c-data-integrity-credential-attachment#credential-offer-exceptions)in this rfc are in line with what is discussed during the vc api calls:

- issuer
- issuanceDate
- credentialSubject.id
- credentialStatus

afaik this is aligning nicely with the direction of the vc-api